neqo-transport/src/send_stream.rs

```diff
@@ -841,6 +847,7 @@ impl SendStream {
                 ref mut send_buf, ..
             } => {
                 send_buf.mark_as_acked(offset, len);
+                self.bytes_acked += len as u64;
```

Comment:
I don't think that this is right.   The spec says:

> Only sequential bytes up to, but not including, the first non-acknowledged byte, are counted. 

You need to access `TxBuffer::retired` for this value.  Unfortunately, this means navigating the `SendStreamState` enum, which doesn't always retain the transmission buffer (or the final size).  Adding a `final_size` to the `DataRecvd` state should cover the success cases.  We will lose the acknowledged data if there is a reset, but you could add another field to each of the `Reset*` states.  The other cases have zero data sent.

Then define a `SendStream::received` function.  That function can check the state and return `retired` if there is a `TxBuffer`, `final_size` if the state is `DataRecvd`, the acknowledged size for `Reset*`, or zero otherwise.

Reply:
I'll try to do this. Thanks.